Makefile

```diff
@@ -0,0 +1,23 @@
+CC := gcc
```

Comment:
Prefer `?=` to allow for values to be overwritten on the command line.

Reply:
Probably should apply to the other variables too, especially `CFLAGS`, `LDFLAGS`.

Reply:
Just to provide modularity/flexability for extensions and/or experimentation !?
Seems like a sound idea.

Reply:
Also because in particular when cross-compiling you might need to set additional flags on those parameters. Thus it's best to only provide sane defaults, but allow the user full flexibility.